### PR TITLE
Fix pe string endianness

### PR DESCRIPTION
**Pull request by wargio**
# DO NOT SQUASH

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixes the string endianness issue on PE files

### CODE REVIEW

File: librz/bin/p/bin_mdmp.c [hunk 1]
```diff
@@ -57,7 +57,7 @@ static RzBinInfo *mdmp_info(RzBinFile *bf) {
 
 	obj = (MiniDmpObj *)bf->o->bin_obj;
 
-	ret->big_endian = obj->endian;
+	ret->big_endian = false;
```
Comment (ret2libc) [inline librz/bin/p/bin_mdmp.c:60]: why?

Comment (wargio) [reply librz/bin/p/bin_mdmp.c:60]: the obj->endian was never initialized. the only code that was initializing it was when `RZ_NEW0(MiniDmpObj)` was called

Comment (wargio) [reply librz/bin/p/bin_mdmp.c:60]: but then it never gets assigned in the whole code.

File: librz/bin/format/pe/pe_rsrc.c [hunk 2]
```diff
@@ -338,7 +338,7 @@ static String *Pe_r_bin_pe_parse_string(RzBinPEObj *bin, PE_DWord *curAddr) {
 		if (*curAddr > bin->size || *curAddr + sizeof(ut16) > bin->size) {
 			goto out_error;
 		}
-		if (!rz_buf_read_le16_at(bin->b, *curAddr, &utf16_char)) {
+		if (rz_buf_read_at(bin->b, *curAddr, (ut8 *)&utf16_char, sizeof(ut16)) != sizeof(ut16)) {
```
Comment (XVilka) [inline librz/bin/format/pe/pe_rsrc.c:341]: Note, that technically UTF16 could be also BE. We need to check if Windows PEs for big-endian platforms still used UTF16LE variant.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/bin/format/pe/pe_rsrc.c (hunk 2 above)
Comment (wargio) [reply librz/bin/format/pe/pe_rsrc.c:341]: no, this is correct because the conversion (including endianness) is done by `rz_str_utf16_to_utf8`

**Pull request merged**